dashdataframe/__init__.py

```diff
@@ -48,104 +102,48 @@ def configure_app(app, df,
-                    id='yaxis-column',
-                    options=[{'label': i, 'value': i}
-                             for i in plot_columns],
-                    value=plot_columns[1]
-                ),
-                dcc.RadioItems(
-                    id='yaxis-type',
-                    options=[{'label': i, 'value': i}
-                             for i in ['Linear', 'Log']],
-                    value='Linear',
-                    labelStyle={'display': 'inline-block'}
-                )
-            ],
-                style={'width': '30%', 'display': 'inline-block'}),
-
-            html.Div([
-                dcc.Dropdown(
-                    id='color_feature',
-                    options=[{'label': i, 'value': i}
-                             for i in color_options],
-                    value='None'
-                )
-            ],
-                style={'width': '30%', 'float': 'right', 'display': 'inline-block'}),
-
-
-            html.Div([
-                html.A(f'{link_name}', id='link', href='', target="_blank")
-            ]),
-
-            dcc.Graph(
-                id='indicator-graphic', style={'width': 1000, 'overflowY': 'scroll'}
-            ),
-
-            html.Div([
-                'Selected Points:  ',
-                html.Span(' Click ', style={'color': 'red'}),
-                html.Span(' Input ID ', style={
-                    'color': 'orange'}),
-                html.Span(' Both ', style={'color': 'orchid'}),
-            ],
-                style={'width': '60%', 'float': 'left', 'display': 'inline-block'}),
-
-            html.Div([
-                dcc.Markdown(dedent('''
-                                    #### Where are these IDs?
-                                    ''')),
-
-                dcc.Input(
-                    id='id_list',
-                    placeholder='Enter list of IDs you would like to highlight',
-                    type='text',
-                    size=100,
-                    value=''
-                )
-            ],
-                style={'width': '100%', 'display': 'inline-block'}),
+    if add_umap is True:
+        print('adding umap tab')
+        umap_layout = dfl.make_umap_layout(plot_columns, link_name)
+        umap_tab = dcc.Tab(label='UMAP Explorer', children=[umap_layout])
+        tabs.append(umap_tab)
+    if add_clustering:
+        print('adding clustering tab')
+        cluster_layout = dfl.make_cluster_layout(plot_columns, link_name)
+        cluster_tab = dcc.Tab(label='Clustering Explorer', children=[cluster_layout])
+        tabs.append(cluster_tab)
 
-            html.Div([
-                html.Button('Digital Sort', id='sortButton'), html.Button(
-                    'Reset Sort', id='resetButton')
-            ]),
-        ])
-    ])
+    app.layout = html.Div([
+    dcc.Tabs(id="tabs", children=tabs,
+        style={'fontSize':'14','fontFamily':'Arial'})
+])
 
+    app._prev_umap_feat_clicks = 0
```

Comment:
shouldn't this be in the if statement? or part of the tab setup? I think you can pass the app to the layout creation.